app/helpers/proxmox_vm_cloudinit_helper.rb

```diff
@@ -41,4 +41,49 @@ def parse_server_cloudinit(args)
     end
     cloudinit_h
   end
+
+  def create_cloudinit_iso(vm_name, configs)
+    iso = File.join('/var/lib/vz/template/iso', "#{vm_name}_cloudinit.iso")
+    arguments = ["-output #{iso}", '-volid', 'cidata', '-joliet', '-rock']
+    ssh = Fog::SSH.new(URI.parse(fog_credentials[:proxmox_url]).host, fog_credentials[:proxmox_username].split('@')[0], { password: fog_credentials[:proxmox_password] })
+    wd = ssh.run([%({mktemp -d})]).first.stdout.chomp
+    configs.each do |config|
+      begin
+        ssh.run([%({echo "#{config[1]}" >> "#{wd}/#{config[0]}"})])
+      rescue ::Foreman::Exception => e
+        logger.warn("Error writing to the file #{config[0]}: #{e}")
+      end
+      arguments.append(File.join(wd, config[0]))
+    end
+    logger.warn("iso image generation args: #{arguments}")
+    raise ::Foreman::Exception, _('ISO build failed') unless ssh.run([%({genisoimage #{arguments.join(' ')}})])
```

Comment:
would be better to collect the output / return code of genisoimage and write a log message accordingly.

Reply:
It gives output like `\nTotal translation table size: 0\r\nTotal rockridge attributes bytes: 250\r\nTotal directory bytes: 0\r\nPath table size(bytes): 10\r\nMax brk space used 0\r\n182 extents written (0 MB)\r\n`  which is not very intuitive for user.